validate.swift

```diff
@@ -2,67 +2,450 @@
 
 import Foundation
 
-// Find the "packages.json" file based on arguments, current directory, or the directory of the script
-let argumentURL = CommandLine.arguments.dropFirst().first.flatMap(URL.init(fileURLWithPath:))
-let currentDirectoryURL = URL(fileURLWithPath: FileManager.default.currentDirectoryPath, isDirectory: true).appendingPathComponent("packages.json")
-let scriptDirectoryURL = URL(fileURLWithPath: #file).deletingLastPathComponent().appendingPathComponent("packages.json")
+// MARK: Configuration Values and Constants
+
+// number of validations to run simultaneously
+let semaphoreCount = 12
+
+let timeoutIntervalForRequest = 3000.0
+let timeoutIntervalForResource = 6000.0
+
+// base url for github raw files
+let rawURLComponentsBase = URLComponents(string: "https://raw.githubusercontent.com")!
+
+// master package list to compare against
+let masterPackageList = rawURLComponentsBase.url!.appendingPathComponent("daveverwer/SwiftPMLibrary/master/packages.json")
```

Comment:
As you don't use `rawURLComponentsBase` anywhere else maybe it would be better to just have the URL in a format like this
 ```suggestion
let masterPackageList = URL(string:"https://raw.githubusercontent.com/daveverwer/SwiftPMLibrary/master/packages.json")!
```

Reply:
Actually I use it when I form the raw GitHub url under the switch case statement:
```swift
  switch host {
  case .GitHub:
    var rawURLComponents = rawURLComponentsBase
    let repositoryName = gitURL.deletingPathExtension().lastPathComponent
    let userName = gitURL.deletingLastPathComponent().lastPathComponent
    rawURLComponents.path = ["", userName, repositoryName, "master", "Package.swift"].joined(separator: "/")
    guard let packageSwiftURL = rawURLComponents.url else {
      return .failure(.invalidURL(gitURL))
    }
    return .success(packageSwiftURL)
  }
```